setup.py

```diff
@@ -64,24 +64,19 @@ def read(*rnames):
         return f.read()
 
 
-Cwrapper = Feature(
-    "C wrapper",
-    standard=True,
-    headers=[os.path.join('src', 'zope', 'proxy', 'proxy.h')],
-    ext_modules=[
-        Extension(
-            "zope.proxy._zope_proxy_proxy",
-            [os.path.join('src', 'zope', 'proxy', "_zope_proxy_proxy.c")],
-        ),
-    ],
-)
+codeoptimization = [
+    Extension(
+        "zope.proxy._zope_proxy_proxy",
+        [os.path.join('src', 'zope', 'proxy', "_zope_proxy_proxy.c")],
+    ),
+]
 
 # PyPy won't build the extension.
 is_pypy = platform.python_implementation() == 'PyPy'
 if is_pypy:
-    features = {}
+    ext_modules = []
```

Comment:
Also, maybe please add an `or is_pure` check with the `os.getenv('PURE_PYTHON')` logic copied from zope.interface etc.?  So people can ask to skip extension modules.

Reply:
Personally, I would like to consider `PURE_PYTHON` at *build* time a deprecated mistake. Runtime, yes, let's be able to select whether to use a C extension or not, but you can't do that if it wasn't even compiled.

Reply:
You're right about this (pip wheel cache poisoning bad!)